src/openrct2/peep/Staff.cpp

```diff
@@ -669,8 +671,8 @@ bool Staff::DoHandymanPathFinding()
         }
     }
 
-    // countof(CoordsDirectionDelta)
-    assert(newDirection < 8);
+    // newDirection can only contain a cardinal direction at this point, no diagonals
+    assert(direction_valid(newDirection));
```

Comment:
This is not correct. `direction_valid()` will return false for directions 4-7, which the old assert considered valid.

Reply:
The old assert was wrong. Firstly, it's not possible for `newDirection` to contain values 4-7 at this point - every assignment to it comes via a codepath which is doing either `& 0x03` or `% NumOrdinalDirections`. Secondly, we are assigning it to PeepDirection below, and if you look through usages of that variable you will see many places where a value greater than 3 would break things.

Reply:
The old assert was to protect invalid array access. This change is correct